x/oracle/keeper/upgrade.go

```diff
@@ -129,3 +129,47 @@ func (k Keeper) SetOracleUpgrade(ctx sdk.Context, upgrade *types.OracleUpgrade)
 	store.Set(key, bz)
 	return nil
 }
+
+func (k Keeper) ApproveOracleUpgrade(ctx sdk.Context, msg *types.MsgApproveOracleUpgrade) error {
+	// validate approval for oracle upgrade
+	if err := k.validateApprovalSharingOracleKey(ctx, msg.GetApprovalSharingOracleKey(), msg.GetSignature()); err != nil {
+		return sdkerrors.Wrapf(types.ErrApproveOracleUpgrade, err.Error())
+	}
+
+	// get oracle upgrade and upgrade info
+	upgradeInfo, err := k.GetOracleUpgradeInfo(ctx)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrApproveOracleUpgrade, err.Error())
+	}
+
+	oracleUpgrade, err := k.GetOracleUpgrade(ctx, upgradeInfo.GetUniqueId(), msg.GetApprovalSharingOracleKey().GetTargetOracleAddress())
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrApproveOracleUpgrade, err.Error())
+	}
+
+	// if EncryptedOraclePrivKey is already set, return error
+	if oracleUpgrade.EncryptedOraclePrivKey != nil {
+		return sdkerrors.Wrapf(types.ErrApproveOracleUpgrade, "already approved oracle upgrade. if you want to be shared oracle private key again, please upgrade oracle again")
+	}
+
+	// update encrypted oracle private key
+	oracleUpgrade.EncryptedOraclePrivKey = msg.GetApprovalSharingOracleKey().EncryptedOraclePrivKey
+
+	// set oracle upgrade
+	if err := k.SetOracleUpgrade(ctx, oracleUpgrade); err != nil {
+		return sdkerrors.Wrapf(types.ErrApproveOracleUpgrade, err.Error())
+	}
+
+	// emit event
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeApproveOracleUpgrade,
+			sdk.NewAttribute(types.AttributeKeyOracleAddress, msg.GetApprovalSharingOracleKey().GetTargetOracleAddress()),
+			sdk.NewAttribute(types.AttributeKeyUniqueID, upgradeInfo.GetUniqueId()),
+		),
+	)
+
+	// TODO: add to queue(?) for update unique ID
```

Comment:
As discussed offline, it seems that you can check if the encryptedOraclePrivKey of OracleUpgrade is not nil without putting it in the queue.

Reply:
@audtlr24, we probably don't need to use queue to update unique ID for upgraded oracles.
As @gyuguen said, we can know that which oracles are upgraded through the `EncryptedOralePrivKey` in `OracleUpgrade`. It makes simpler I think.

Reply:
Okay, I understand. I will reflect it in PR #584.